apps/storefront/src/pages/QuoteDetail/index.tsx

```diff
@@ -269,28 +351,40 @@ function QuoteDetail() {
       oos: oosErrorList,
       nonPurchasable: nonPurchasableErrorList,
     });
+  };
+
+  const validateQuoteProducts = isMoveStockAndBackorderValidationToBackend
+    ? quoteDetailBackendEndValidations
+    : quoteDetailFrontEndValidations;
+
+  useEffect(() => {
+    validateQuoteProducts();
     // disabling since b3Lang is a dependency that will trigger rendering issues
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [isEnableProduct, isHandleApprove, productList]);
 
   const proceedingCheckoutFn = useCallback(() => {
-    if (isHideQuoteCheckout) {
-      const { oos, nonPurchasable } = noBuyerProductName;
-      if (oos)
-        snackbar.error(
-          b3Lang('quoteDetail.message.insufficientStock', {
-            ProductName: oos,
-          }),
-        );
-
-      if (nonPurchasable)
-        snackbar.error(
-          b3Lang('quoteDetail.message.nonPurchasable', {
-            ProductName: nonPurchasable,
-          }),
-        );
+    if (!isMoveStockAndBackorderValidationToBackend) {
```

Comment:
follow the same pattern as with `validateQuoteProducts`, two methods (even if one is simply a `() => true`